test/src/cli/flutter_cli_test.dart

```diff
@@ -168,25 +168,62 @@ void main() {
         );
       });
 
-      test('completes when there is a pubspec.yaml (recursive)', () {
-        final directory = Directory.systemTemp.createTempSync();
-        final nestedDirectory = Directory(p.join(directory.path, 'test'))
-          ..createSync();
-        File(p.join(nestedDirectory.path, 'pubspec.yaml'))
-            .writeAsStringSync(_pubspec);
-
-        ProcessOverrides.runZoned(
-          () => expectLater(
-            Flutter.packagesGet(
-              cwd: directory.path,
-              recursive: true,
-              logger: logger,
+      test(
+        'completes when there is a pubspec.yaml and '
+        'directory is ignored (recursive)',
+        () {
+          final directory = Directory.systemTemp.createTempSync();
+          final nestedDirectory = Directory(p.join(directory.path, 'test'))
+            ..createSync();
+          final ignoredDirectory = Directory(
+            p.join(directory.path, 'test_plugin'),
+          )..createSync();
+          final ignoredDirectoryTwo = Directory(
+            p.join(directory.path, 'test_plugin_two'),
+          )..createSync();
+          final ignoredDirectoryThree = Directory(
+            p.join(directory.path, 'test_plugin_three'),
+          )..createSync();
+          final ignoredDirectoryFour = Directory(
+            p.join(directory.path, 'test_plugin_four'),
+          )..createSync();
+
+          File(p.join(nestedDirectory.path, 'pubspec.yaml'))
+              .writeAsStringSync(_pubspec);
+          File(p.join(ignoredDirectory.path, 'pubspec.yaml'))
+              .writeAsStringSync(_pubspec);
+          File(p.join(ignoredDirectoryTwo.path, 'pubspec.yaml'))
+              .writeAsStringSync(_pubspec);
+          File(p.join(ignoredDirectoryThree.path, 'pubspec.yaml'))
+              .writeAsStringSync(_pubspec);
+          File(p.join(ignoredDirectoryFour.path, 'pubspec.yaml'))
+              .writeAsStringSync(_pubspec);
+
+          ProcessOverrides.runZoned(
+            () => expectLater(
+              Flutter.packagesGet(
+                cwd: directory.path,
+                recursive: true,
+                ignore: {
+                  'test_plugin',
+                  '/**/test_plugin_two/**',
+                  '${directory.path}/test_plugin_three',
+                  '${directory.path}/test_plugin_four/'
+                },
+                logger: logger,
+              ),
+              completes,
             ),
-            completes,
-          ),
-          runProcess: process.run,
-        );
-      });
+            runProcess: process.run,
+          ).whenComplete(() {
+            verify(() {
```

Comment:
Shouldn't we verify that we're running flutter packages get in the specific packages? The current verify isn't very specific and this test would likely pass even without the changes in the branch.